PowerControl/Helpers/ProfilesController.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CommonHelpers;
+using PowerControl.Helper;
+using PowerControl.Helpers;
+using PowerControl.Menu;
+
+namespace PowerControl.Helpers
+{
+    public static class ProfilesController
+    {
+        private const string FpsKey = "FPSLimit";
+        private const string RefreshRateKey = "RefreshRate";
+        private const string IsTroubledKey = "IsTroubled";
+        private const string DefaultName = "Default";
+
+        private static string CurrentGame = string.Empty;
+        private static ProfileSettings DefaultSettings = new ProfileSettings(DefaultName);
+        private static ProfileSettings? CurrentSettings;
+        private static string[] troubledGames = { "dragonageinquisition" };
+        private static bool IsWriteLocked = false;
+
+        private static System.Windows.Forms.Timer? timer; 
+
+        public static void Initialize()
+        {
+            Options.RefreshRate.Instance?.SetValueChanged((_, _, newValue) =>
```

Comment:
I think this is wrong. There's nothing special about FPS Limit or Refresh Rate. I would propose instead that to `MenuWithItemOptions` we add `public string? PersistenceKey { get; set; }` and set that for every property that needs to be persisted. Then we do something like this:

```csharp
public class ProfilesController
{
    private const string DefaultName = "Default";

    private ProfileSettings DefaultSettings = new ProfileSettings(DefaultName);
    private ProfileSettings? CurrentSettings;

    private System.Windows.Forms.Timer timer = new System.Windows.Forms.Timer();

    public void ProfilesController()
    {
        MenuRoot.Stack += Menu_ValueChanged;

        timer = new System.Windows.Forms.Timer();
        timer.Interval = 1000;
        timer.Tick += delegate
        {
          timer.Enabled = false;
          try { Tick(); }
          finally { timer.Enabled = true; }
        };
        timer.Start();
    }
    
    private void Menu_ValueChanged(MenuItemWithOptions options, String oldValue, String newValue)
    {
      if (options.PersistenceKey is null)
        return;
      
      CurrentSettings.Set<string>(options.PersistenceKey, newValue);
    }

    private void OnProfileChanged(String newProfile)
    {
      CurrentSettings = new ProfileSettings(newProfile);

      foreach(var options in MenuStack.Root.AllMenuItemOptions())
      {
        if (options.PersistenceKey is null)
          continue;

        var value = CurrentSettings.Get<string>(options.PersistenceKey, "");
        if (value == "")
          continue;

        // TODO: maybe for cases where settings needs to be applied are delayed
        // We instead of false pass a delay duration
        try { options.Set(value, false); }
        catch(Exception e) { TraceException("Profiles", newProfile, e); }
      }
    }
}
```

I made also a few more changes:
- We should initialize `ProfilesController`, so not static class
- It uses hook onto MenuStack
- It iterates MenuItems to set settings
- Gets rid of special write locked, and delay handling - instead if this is needed it should rather be part of the `Item.Set(,..., delay)`
- Does not include those other changes mentioned about process discovery

Reply:
I like the idea of having keys inside menu items. I used fps and refresh rate for now, because I'm developing it on my laptop and also these two options are the most wanted in profiles, so I want to first make MVP with a good base which later on can be extended.

I will implement dynamic menu options handling, but probably write lock and delay will stay as they were. Lock is there, because change to refresh rate resets fps and it's a good thing when user changes it, but it's a bad thing when program set's it. I have an idea to split ApplyValue into ApplyValue and ApplyCleanUp and former will be used if some `silent` parameter will be set, but I have to try it first.
Delay on the other hand is crucial in it's current form. You can't delay just one option - bc refresh rate, as said before, resets fps. So you have to delay them all. Also bc thread is sleeping it makes sure OSD is "frozen" during the load and user can't change anything.
If my change from ApplyValue into ApplyValue and ApplyCleanUp will succeed and after other things we can get back to this one.

And for now I won't implement the new Tick, unless there will be working version of it. I will try to figure something on my own, too.